### ISSUE TITLE
'is' operator type check

### ISSUE THREAD

**Issue by Akuli** [python/mypy#10166]
**Feature**

`is` operator type check

**Pitch**

`"a" is 1` should be type error

**Comment by JukkaL** [comment]
This is already caught by mypy if you use `--strict-equality`. However, maybe we should turn this on by default, since it's quite useful.

**Comment by JukkaL** [comment]
Added #10168 about turning this on by default.

**Comment by MetRonnie** [comment]
I don't get a Non-overlapping equality error for `if value is None` here:
```python
def foo(value: str) -> Optional[str]:
    if value is None:
        return None
    return value
```

https://mypy-play.net/?mypy=latest&python=3.7&flags=ignore-missing-imports%2Cstrict&gist=3ac2aae64989362ea904cc3058ea854c

**Comment by MetRonnie** [comment]
Although I suppose that could be intentional, as you might want to do `if x is None: raise TypeError`

**Comment by JukkaL** [comment]
> Although I suppose that could be intentional

Yeah, this is intentional since redundant `None` checks tend to be quite common and could otherwise cause false positives.
